src/checkFirebaseSDKVersion.ts

```diff
@@ -0,0 +1,96 @@
+import * as _ from "lodash";
+import * as clc from "cli-color";
+import * as path from "path";
+import * as semver from "semver";
+import * as spawn from "cross-spawn";
+import * as utils from "./utils";
+import * as logger from "./logger";
+
+interface NpmListResult {
+  name: string;
+  dependencies: {
+    "firebase-functions": {
+      version: string;
+      from: string;
+      resolved: string;
+    };
+  };
+}
+
+interface NpmShowResult {
+  "dist-tags": {
+    latest: string;
+  };
+}
+
+/** Returns the version of firebase-functions SDK specified by package.json and package-lock.json.
+ * @param sourceDir Source directory of functions code
+ **/
+export function getFunctionsSDKVersion(sourceDir: string): string | void {
+  try {
+    const child = spawn.sync("npm", ["list", "firebase-functions", "--json=true"], {
+      cwd: sourceDir,
+      encoding: "utf8",
+    });
+    if (child.error) {
+      logger.debug(child.error.stack);
+      return;
+    }
+    const output: NpmListResult = JSON.parse(child.stdout);
+    return _.get(output, ["dependencies", "firebase-functions", "version"]);
+  } catch (e) {
+    logger.debug("getFunctionsSDKVersion encountered error:", e);
+    return;
+  }
+}
+
+/**
+ * Checks if firebase-functions SDK is not the latest version in NPM, and prints update notice if it is outdated.
```

Comment:
Given the way this is written, I would say something like "if unable to determine the local or remote version, this does nothing"